subsys/bluetooth/host/l2cap.c

```diff
@@ -1003,14 +1004,26 @@ static uint16_t l2cap_chan_accept(struct bt_conn *conn,
 	return BT_L2CAP_LE_SUCCESS;
 }
 
-static bool l2cap_check_security(struct bt_conn *conn,
+static int l2cap_check_security(struct bt_conn *conn,
 				 struct bt_l2cap_server *server)
 {
 	if (IS_ENABLED(CONFIG_BT_CONN_DISABLE_SECURITY)) {
-		return true;
+		return 0;
 	}
 
-	return conn->sec_level >= server->sec_level;
+	if (conn->sec_level < server->sec_level) {
+		if (server->sec_level == BT_SECURITY_L2) {
+			return BT_L2CAP_LE_ERR_ENCRYPTION;
+		} else if (server->sec_level == BT_SECURITY_L3) {
+			if (IS_ENABLED(CONFIG_BT_SMP_ENFORCE_MITM)) {
+				return BT_L2CAP_LE_ERR_AUTHORIZATION;
+			} else {
+				return BT_L2CAP_LE_ERR_AUTHENTICATION;
+			}
```

Comment:
I don't think this part is right. Are there tests expecting BT_L2CAP_LE_ERR_AUTHORIZATION error code?

Authorization is asking the user for confirmation before continuing, returning it here based on ENFORCE_MITM is wrong.

Reply:
Tests expecting this error are L2CAP/LE/CFC/BV-13-C and L2CAP/ECFC/BV-13-C  (for ATT and EATT, respectively)